src/main/java/com/mathworks/ci/MatlabBuildStepExecution.java

```diff
@@ -0,0 +1,107 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2022 The MathWorks, Inc.
+ *  
+ */
+
+import java.io.IOException;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Launcher.ProcStarter;
+import hudson.model.Result;
+import hudson.model.TaskListener;
+
+public class MatlabBuildStepExecution extends SynchronousNonBlockingStepExecution<Void> implements MatlabBuild {
+    
+    private static final long serialVersionUID = 6704588180717665100L;
+    
+    private String tasks;
+
+    public MatlabBuildStepExecution(StepContext context, String tasks) {
+        super(context);
+        this.tasks = tasks;
+    }
+
+    private String getTasks() {
+        return this.tasks;
+    }
+
+    @Override
+    public Void run() throws Exception {
+        final Launcher launcher = getContext().get(Launcher.class);
+        final FilePath workspace = getContext().get(FilePath.class);
+        final TaskListener listener = getContext().get(TaskListener.class);
+        final EnvVars env = getContext().get(EnvVars.class);
+        
+        //Make sure the Workspace exists before run
+        
+        workspace.mkdirs();
+        
+        int exitCode = execMatlabCommand(workspace, launcher, listener, env);
+
+        if(exitCode != 0){
+            // throw an exception if return code is non-zero
+            stop(new MatlabExecutionException(exitCode));
+        }
+
+        getContext().setResult(Result.SUCCESS);
+        return null;
+    }
+
+    @Override
+    public void stop(Throwable cause) throws Exception {
+        getContext().onFailure(cause);
+    }
+    
+    private int execMatlabCommand(FilePath workspace, Launcher launcher,
```

Comment:
This is almost identical in several files. Might be a way to abstract or reuse in the future

Reply:
Yeah we could declare in MatlabBuild interface